src/core/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,370 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { Input, InputProps, Menu, MenuItem, SelectOption, Text } from '../..';
+import {
+  InputContainer,
+  useTheme,
+  Popover,
+  PopoverProps,
+  CommonProps,
+  getFocusableElements,
+  getRandomValue,
+} from '../utils';
+import SvgCaretDownSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretDownSmall';
+import 'tippy.js/animations/shift-away.css';
+
+export type ComboBoxProps<T> = {
+  /**
+   * Array of options that populate the dropdown list.
+   */
+  options: SelectOption<T>[];
+  /**
+   * Controlled value of ComboBox.
+   */
+  value?: T;
+  /**
+   * Callback fired when selected value changes.
+   */
+  onChange?: (value: T) => void;
+  /**
+   * Function to customize the default filtering logic.
+   */
+  filterFunction?: (
+    options: SelectOption<T>[],
+    inputValue: string,
+  ) => SelectOption<T>[];
+  /**
+   * Native input element props.
+   */
+  inputProps?: Omit<InputProps, 'setFocus'>;
+  /**
+   * Props to customize dropdown menu behavior.
+   */
+  dropdownMenuProps?: PopoverProps;
+  /**
+   * Message shown when no options are available (after filtering).
```

Comment:
Not only after filtering because if initial options are empty too, the same message will be shown.

Reply:
Fixed